src/registryctl/api/registry/manifest/manifest.go

```diff
@@ -0,0 +1,58 @@
+package manifest
+
+import (
+	"github.com/docker/distribution/registry/storage"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/registryctl/api"
+	regConf "github.com/goharbor/harbor/src/registryctl/config/registry"
+	"github.com/gorilla/mux"
+	"github.com/opencontainers/go-digest"
+	"net/http"
+	"strings"
+)
+
+// NewHandler returns the handler to handler manifest request
+func NewHandler() http.Handler {
+	return &handler{}
+}
+
+type handler struct{}
+
+// ServeHTTP ...
+func (h *handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	switch req.Method {
+	case http.MethodDelete:
+		h.delete(w, req)
+	default:
+		api.HandleForbidden(w)
+	}
+}
+
+// delete deletes manifest ...
+func (h *handler) delete(w http.ResponseWriter, r *http.Request) {
+	ref := mux.Vars(r)["reference"]
+	if ref == "" {
+		api.HandleBadRequest(w, errors.New("no reference specified"))
+		return
+	}
+	dgst, err := digest.Parse(ref)
+	if err != nil {
+		api.HandleBadRequest(w, errors.Wrap(err, "not supported reference"))
+		return
+	}
+	repoName := mux.Vars(r)["name"]
+	if repoName == "" {
+		api.HandleBadRequest(w, errors.New("no repository name specified"))
+		return
+	}
+	var tags []string
+	v := r.URL.Query()
+	queryTags := v.Get("tags")
```

Comment:
Do we need tags here? Since 2.0, tags may exist in harbor db but not exist in registry.

Reply:
the fundamental method needs this pararmeter, and it can be a null. Leave it at here will not impact GC deletes manifest via digest only.

Sometime, we may need to parse the tag into delete method as in Harbor v2.0.0, the first tag of artifact will be in the backend storage。 -- But, this brings another issue, how can GC know which is the first tag of the artifact in Harbor? otherwise, we have to leveage the code to enumerate repository tags.

Reply:
I feel this may reflect a misunderstanding here, I don't think the parm  "tags" makes sense for this API...

Reply:
ok, I will remove it.